x/stakeibc/keeper/icqcallbacks_withdrawal_balance.go

```diff
@@ -8,119 +8,118 @@ import (
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
 	"github.com/spf13/cast"
 
+	"github.com/Stride-Labs/stride/v4/utils"
 	icqtypes "github.com/Stride-Labs/stride/v4/x/interchainquery/types"
 	"github.com/Stride-Labs/stride/v4/x/stakeibc/types"
 )
 
 // WithdrawalBalanceCallback is a callback handler for WithdrawalBalance queries.
+// The query response will return the withdrawal account balance
+// If the balance is non-zero, ICA MsgSends are submitted to transfer from the withdrawal account
+//  to the delegation account (for reinvestment) and fee account (for commission)
 // Note: for now, to get proofs in your ICQs, you need to query the entire store on the host zone! e.g. "store/bank/key"
 func WithdrawalBalanceCallback(k Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
```

Comment:
after refactoring this, any thoughts on whether we can/should modularize it more? it's already a bit cleaner with sdk.Ints, but still pretty hefty

Reply:
Good question! I could probably go either way. My two cents:

I think functions like this where the bulk of the function just checks error cases (e.g. make sure host exists, make sure ICAs are initialzed, etc.) are lower priority refactors. 
That said, I think we should refactor all functions that submit ICAs in such a way that there's a function that first returns just the ICA message itself, and then a separate function that actually executes it. That way we can write a test that checks the contents of the message